### DISCUSSION TITLE
2.0.1

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#10538] [General]
*  Mount cgroupv2 at /sys/fs/cgroup if cgroupv1 is disabled (solves #10497)
*  Conditionally write the explorer extensions registry keys (solves #9231)
*  Don't try to inherit the same handle twice (solves #10488)
*  Fix parsing of '0GB' memory strings (solves #10503)
*  Update WSLg to [1.0.58](https://github.com/microsoft/wslg/releases/tag/v1.0.58)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/2.0.1'>2.0.1</a>.</em>

**Comment by eur2fe** [comment]
I tried to update from 2.0.0 to 2.0.1 and it does not work. No error message. Tried it on two computers, it does something, but WSL stays on 2.0.0.

This code was run on Windows Server 2022 (as a VM via VMware):

```console
PS C:\Windows\system32> &"C:\Program Files\WSL\wsl.exe" --version
WSL version: 2.0.0.0
Kernel version: 5.15.123.1-1
WSLg version: 1.0.57
MSRDC version: 1.2.4485
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25880.1000-230602-1350.main
Windows version: 10.0.20348.1970
PS C:\Windows\system32> &"C:\Program Files\WSL\wsl.exe" --update --pre-release
Checking for updates.
Updating Windows Subsystem for Linux to version: 2.0.1.
PS C:\Windows\system32> &"C:\Program Files\WSL\wsl.exe" --version
WSL version: 2.0.0.0
Kernel version: 5.15.123.1-1
WSLg version: 1.0.57
MSRDC version: 1.2.4485
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25880.1000-230602-1350.main
Windows version: 10.0.20348.1970
PS C:\Windows\system32>
```

**Reply by TerrorJack** [reply]
same on 22621.2361. `wsl --update --pre-release` seems to be a no-op in a privileged shell. the command used to work in an unprivileged shell though, but it fails with:

```
Update failed (exit code: 1603).
Log file: C:\Users\TERROR~1\AppData\Local\Temp\wsl-install-logs.txt
Error code: Wsl/UpdatePackage/E_ABORT
```

**Comment by blakeduffey** [comment]
I asked a question in a different thread about dueling versions of wsl.exe (which was ignored)

When I log in with my non-privileged user, `wsl --version` returns 2.0.0.0
When I log in with my privileged user and run `wsl --version` - I get a usage message

My guess is the privileged user is referencing a different version.  There is nothing in the user path for the non-privileged user that would point to C:\Program Files\WSL.

FWIW - I've not use the 'store' at all - I've simply issued `wsl --install`.  Updates previously didn't require admin, but now we are told it does.  If user have multiple accounts (non-priv vs priv) - this might be part of the confusion.
